src/pulse3D/peak_detection.py

```diff
@@ -450,3 +448,29 @@ def add_levels(df):
 
     dfs = [add_levels(df) for df in dfs]
     return pd.concat(dfs, axis=axis, *args, **kwargs)
+
+
+def get_windowed_peaks_valleys(
+    start_idx: int, end_idx: int, peaks: NDArray, valleys: NDArray
+) -> Tuple[NDArray, NDArray]:
+    """Convert peaks and valleys to fit window of analysis.
+
+    Args:
+        start_idx: int start index to window of analysis to substract from peak/valley indices
+        end_idx: int end index to window of analysis to contain only relevant peak/valley indices
+        peaks: NDArray contains new peak indices to fit window of analysis
+        valleys: NDArray contains new valley indices to fit window of analysis
+
+    Returns:
+        peaks: NDArray of peak indices
+        valleys: NDArray of valley indices
+    """
+    windowed_indices = end_idx - start_idx
+    # remove up to starting index to make start time index 0
+    sub_peaks = np.subtract(peaks, start_idx)
+    sub_valleys = np.subtract(valleys, start_idx)
+    # remove indices greater than max windowed index
+    filtered_peaks = np.where((sub_peaks < windowed_indices) & (sub_peaks >= 0))[0]
+    filtered_valleys = np.where((sub_valleys < windowed_indices) & (sub_valleys >= 0))[0]
+    # grab indices from original sub peaks and valleys
+    return (sub_peaks[filtered_peaks], sub_valleys[filtered_valleys])
```

Comment:
nitpick, don't need parens here

Reply:
truuuuuueeee, did that when it was returning something else and didn't swap back